src/chainRec.js

```diff
@@ -0,0 +1,35 @@
+var _arrayChainRec = require('./internal/_arrayChainRec');
+var _curry3 = require('./internal/_curry3');
+
+
+/**
+ * `chainRec` does stack safe recurtion on List
+ *
+ * Dispatches to the `chainRec` method of first argument, if present,
+ * according to the [FantasyLand ChainRec spec](https://github.com/fantasyland/fantasy-land#chainrec).
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig ChainRec m => m -> ((a -> c, b -> c, a) -> m c) -> a -> m b
+ * @param {ChainRec} object or type which is ChainRec
+ * @param {Function} fn The function to compute next step of recursion
+ * @param {Any} initial value for computation
+ * @return {Array} The result of recursive computation
+ * @example
+ *
+ *      R.chainRec(Array, (next, done, a) => (
+ *        a.length == 2 ? [a + '!', a + '?'].map(done)
+ *                      : [a + 'a', a + 'b'].map(next)
+ *      ), ''); // [ 'aa!', 'aa?', 'ab!', 'ab?', 'ba!', 'ba?', 'bb!', 'bb?' ],
+ *
+ */
+module.exports = _curry3(function _chainRec(chainRec, fn, i) {
+  if (chainRec != null && typeof chainRec.chainRec === 'function') {
```

Comment:
Why `chainRec != null`? `chainRec` should never be `null` or `undefined`, surely? Having `R.chainRec(null)` behave like `R.chainRec([])` seems strange to me.